Tests/StoreKitUnitTests/PurchasesOrchestratorTests.swift

```diff
@@ -489,6 +578,85 @@ class PurchasesOrchestratorTests: StoreKitConfigTestCase {
         expect(self.backend.invokedPostReceiptDataParameters?.offeringIdentifier) == "offering"
     }
 
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, *)
+    func testPurchaseSK2PackageDoesNotPostAdServicesTokenIfNotEnabled() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+        try AvailabilityChecks.skipIfTVOrWatchOS()
+
+        let mockListener = try XCTUnwrap(
+            self.orchestrator.storeKit2TransactionListener as? MockStoreKit2TransactionListener
+        )
+
+        self.attributionFetcher.adServicesTokenToReturn = "token"
+        self.customerInfoManager.stubbedCachedCustomerInfoResult = self.mockCustomerInfo
+        self.backend.stubbedPostReceiptResult = .success(self.mockCustomerInfo)
+        mockListener.mockTransaction = .init(try await self.simulateAnyPurchase())
+
+        let product = try await self.fetchSk2Product()
+
+        let package = Package(identifier: "package",
+                              packageType: .monthly,
+                              storeProduct: StoreProduct(sk2Product: product),
+                              offeringIdentifier: "offering")
+
+        _ = try await self.orchestrator.purchase(sk2Product: product, package: package, promotionalOffer: nil)
+
+        expect(self.receiptFetcher.receiptDataCalled) == true
+        expect(self.receiptFetcher.receiptDataReceivedRefreshPolicy) == .always
+
+        expect(self.backend.invokedPostReceiptDataCount) == 1
+        expect(self.backend.invokedPostReceiptDataParameters?.aadAttributionToken).to(beNil())
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, *)
+    func testPurchaseSK2PackagePostsAdServicesTokenAndAttributes() async throws {
```

Comment:
feels like we need another set of 2 tests for "doesn't post the token if it's already been posted, sk1 / sk2"

Reply:
I suppose that's tested through AttributionPoster, though, right?

Reply:
Tested by `testAdServicesTokenToPostIfNeededReturnsNilIfAlreadySent`